src/libecalc/core/graph_result.py

```diff
@@ -724,50 +850,124 @@ def get_asset_result(self) -> libecalc.dto.result.EcalcModelResult:
+                        consumer_result.component_result.power, regularity=regularity
+                    )
+                    .to_volumes()
+                    .to_unit(Unit.GIGA_WATT_HOURS)
+                    .cumulative()
+                    .dict()
+                    if consumer_result.component_result.power is not None
+                    else None,
+                    "power": TimeSeriesRate.from_timeseries_stream_day_rate(
+                        consumer_result.component_result.power, regularity=regularity
+                    ),
+                    "energy_usage": TimeSeriesRate.from_timeseries_stream_day_rate(
+                        consumer_result.component_result.energy_usage, regularity=regularity
+                    ).to_calendar_day()
+                    if consumer_result.component_result.energy_usage.unit == Unit.STANDARD_CUBIC_METER_PER_DAY
+                    else TimeSeriesRate.from_timeseries_stream_day_rate(
+                        consumer_result.component_result.energy_usage, regularity=regularity
+                    ).to_stream_day(),
+                },
             )
+            if isinstance(obj, GeneratorSetResult):
+                obj.power_capacity_margin = TimeSeriesRate.from_timeseries_stream_day_rate(
+                    consumer_result.component_result.power_capacity_margin, regularity=regularity
+                )
+            elif isinstance(obj, PumpResult):
+                obj.inlet_liquid_rate_m3_per_day = TimeSeriesRate.from_timeseries_stream_day_rate(
+                    consumer_result.component_result.inlet_liquid_rate_m3_per_day, regularity=regularity
+                )
+                obj.inlet_pressure_bar = consumer_result.component_result.inlet_pressure_bar
+                obj.outlet_pressure_bar = consumer_result.component_result.outlet_pressure_bar
+                obj.operational_head = consumer_result.component_result.operational_head
+
+            elif isinstance(obj, CompressorResult):
+                obj.recirculation_loss = TimeSeriesRate.from_timeseries_stream_day_rate(
+                    consumer_result.component_result.recirculation_loss, regularity=regularity
+                )
+                obj.rate_exceeds_maximum = consumer_result.component_result.rate_exceeds_maximum
+                obj.outlet_pressure_before_choking = consumer_result.component_result.outlet_pressure_before_choking
+            elif isinstance(obj, AssetResult):
+                obj.hydrocarbon_export_rate = TimeSeriesRate.from_timeseries_stream_day_rate(
+                    consumer_result.component_result.hydrocarbon_export_rate, regularity=regularity
+                )
+                obj.emission_intensities = consumer_result.component_result.emission_intensities
+            elif isinstance(obj, InstallationResult):
+                obj.regularity = consumer_result.component_result.regularity
```

Comment:
TODO: don't use parse_obj above, instead create the object after the instance check core resultcomponent result?